sammba/modality_processors/perfusion_fair.py

```diff
@@ -0,0 +1,369 @@
+    ----------
+    s0 : numpy array of int or float
+        The aquired signals.
+        
+    ti : list of int or float
+        The inversion times. Must have the same length as s0.
+        
+    t1_guess : int or float
+        An initial starting value for T1. The mean of TIs is a good guess.
+
+    Returns
+    -------
+    class scipy.optimize.OptimizeResult    
+    
+    Notes
+    ----- 
+    Levenberg-Marquardt (method='lm'), the usual literature algorithm, cannot
+    be used here as the implementation in scipy does not accept bounds. From
+    experience with the R MINPACK implementation (minpack.lm function nlsLM),
+    these bounds are necessary to assure good fitting. The lmfit python package
+    does implement bounds for MINPACK Levenberg-Marquardt (method=leastsq;
+    default), though the results appear to be worse than for TRR
+    (method=least_squares), whose results bizarrely look closer to those of R
+    minpack.lm nlsLM (to confirm), and are of course identical to
+    scipy.optimize.least_squares with method='trf' (default).
+    """
+    
+    return ls(_fair_t1_func, np.array([0, np.mean(s0), t1_guess]),
+              bounds=([0, 0, 0], np.inf), args=(np.array(ti), s0))
+
+
+def perf_fair_fitter(s0, t1_blood, lambda_blood, ti, multiplier, t1_guess,
+                     picker_sel, picker_nonsel, outtype='simple'):
+    """
+    Wrapper to execute fair_t1_fit on a real signal vector containing both
+    selective and non-selective inversions. Also calculates rCBF and absolute
+    CBF from the resulting selective and non-selective T1 values.
+    
+    Parameters
+    ----------
+    s0 : numpy array of int or float
+        The aquired signals.
+        
+    t1_blood : int or float
+        T1 of blood in ms at the acquisition field strength. Empirical
+        constant; an example value is 2800 at 11.7T (Duong ref).
+        
+    lambda_blood : int or float
+        The assumed blood tissue partition coefficient of water in ml per g.
+        Empirical constant; usually 0.9.
+        
+    ti : list of int or float
+        The inversion times. Must have the same length as s0.
+        
+    multiplier : int or float
+        The absolute CBF result is initially produced in units of ml per g per
+        ms. The multiplier converts to desired units. Usually it is 6000000 to
+        convert to ml per 100g per min.    
+        
+    t1_guess : int or float
+        An initial starting value for T1. The mean of TIs is a good guess.
+    
+    picker_sel : list or numpy array of int
+        Vector indicating positions of selectively inverted signals in s0. The
+        lengths of picker_sel and picker_sel must sum to the length of s0.
+        
+    picker_nonsel : list or numpy array of int
+        Vector indicating positions of non-selectively inverted signals in s0.
+        The lengths of picker_sel and picker_sel must sum to the length of s0.
+        
+    outtype : {'simple', 'complicated'}, optional

```

Comment:
I always prefer having one unique output type if it is possible

Reply:
it's useful for diagnostic use; in any case this function may be worth unexposing